### ISSUE TITLE
Turn deprecation warning into error for .gas and .value

### ISSUE THREAD

**Issue by leonardoalt** [ethereum/solidity#8147]

**Comment by chriseth** [comment]
The alternative is fully implemented now.

**Comment by ekpyron** [comment]
To elaborate a bit:
We recently introduced new syntax for specifying gas and value for external function calls. The old syntax was:
```
contract C {
    function g() external payable {}
    function f() public payable {
        this.g.value(42).gas(23)();
    }
}
```
The new syntax is:
```
contract C {
    function g() external payable {}
    function f() public payable {
        this.g{value: 42, gas: 23}();
    }
}
```
Similarly, the syntax for contract creation ``new C.value(42)()`` was replaced by ``new C{ value: 42}()``.
While we need to support the old syntax until the next breaking release, we should add a deprecation warning for it now.
The first part for function calls will entail modifying ``TypeChecker::visit(MemberAccess const&)`` to emit a warning along the lines of "<function>.value(...)() is deprecated. Use <function>{value: ...}() instead." (we can discuss the precise wording) and to update the tests accordingly. Contract creation we can either handle in the same or in a separate PR.
